docs/mpid_messaging.md

```diff
@@ -61,13 +63,31 @@ Mpid (A) -> - *                                    * - <-Mpid (B)
 
 ```
 1. The user at Mpid(A) sends MpidMessage to MpidManager(A) signed with the recipient included
-2. The MpidManagers(A) sync this message and perform the action() which sends the MpidAlert to MpidManagers(B) [the ```MpidAlert::message_id``` at this stage is simply the hash of the MpidMessage.
-3. MpidManager(B) either store the MpidAlert or send immediately to the Mpid(B) user if they are off-line or on-line respectively.
-4. Mpid(B) then sends a ```Signed(retrieve_message)``` to the MpidManager(B) group who send this on to the MpidManagers(A). This message is of the form ```retrieve_message(MpidAlert, MpidPacket)[signed by Mpid(B)]``` 
-5. MpidManagers(A) then sync() the MpidAlert and confirm this is from the MpidManager(B) group, do a signature check and then perform action() which sends the message to MpidManagers(B) and remove the message.
-6. MpidManager(B) then sync() the message to confirm it was delivered from MpidManagers(A) and send the message to Mpid(B), or store for later retieval if the node has gone off-line. 
+2. The MpidManagers(A) sync this message and perform the action() which sends the MpidHeader to MpidManagers(B) [the ```MpidAlert::message_id``` at this stage is hash of the MpidMessage.
+3. MpidManager(B) stores the MpidAlert and sends the alert to Mpid(B) as soon as it is found online.
+4. On receving the alert, Mpid(B) sends a ```retrieve_message``` to MpidManagers(B) which is forwarded to MpidManagers(A). This message is of the form ```retrieve_message(MpidAlert, MpidPacket)``` 
+5. MpidManagers(A) sends the message to MpidManagers(B) which is forwarded to MPid(B) if MPid(B) is online.
+6. On receiving the message, Mpid(B) sends a remove request to MpidManagers(B), MpidManagers(B) sync remove the corresponding alert and forward the remove request to MpidManager(A). MpidManagers(A) sync remove the corresponding entry. 
+7. When Mpid(A) decides to remove the MpidMessage from the OutBox, if the message hasn't been retrived by Mpid(B) yet. The MpidManagers(A) group should not only remove the correspondent MpidMessage from their OutBox of Mpid(A), but also send a notification to the group of MpidManagers(B) so they can remove the correspodent MpidAlert from their InBox of Mpid(B).
+8. When Mpid(B) send a request to fetch message head only, it will directly goes to MpidManagers(A) and get response. This will not trigger the removal of such message in MpidManagers(A).
```

Comment:
if message_header become part of the MpidAlert, then this step 8 is not nessarty